File: Tier1/HttpClients/ClientImplementations/HousingInterface.cs
using Shared.DTOs;

namespace HttpClients.ClientImplementations;

public interface HousingInterface
{
    Task<HousingDTO> AddHousingAsync(HousingCreationDTO dto);
    Task<HousingListDTO> GetAvailableHousing();
    Task<HousingDTO> RemoveHousing(HousingIdDTO dto);
}

File: Tier2/src/main/java/via/sep3/group11/tier2/logicInterfaces/HousingInterface.java
package via.sep3.group11.tier2.logicInterfaces;

import via.sep3.group11.tier2.shared.DTOs.*;
import via.sep3.group11.tier2.shared.domain.Housing;
import via.sep3.group11.tier2.shared.exceptions.ValidationException;

public interface HousingInterface {
    HousingDTO addHousing(HousingCreationDTO dto);
    HousingListDTO getAvailableHousing();
    HousingDTO removeHousing(HousingIdDTO dto);
    HousingListDTO getHousingByHostId(String email);


}
